### PR TITLE
feature/CD-78: implemented images search by text

### PR DESCRIPTION
**Pull request by gusartus**

### CODE REVIEW

File: src/common/selectors/entities.ts [hunk 1]
```diff
@@ -10,6 +11,16 @@ export const getBuildingImages = createSelector(
 	images => images.filter((item: any) => item.build.status === 'building'),
 );
 
+export const getFilteredImages = createSelector(
+	getImages,
+	getImagesTextFilter,
+	(images, substr) => images.filter((image: any) => (
+		image.notes.toLowerCase().includes(substr.toLowerCase())
+		|| image.distro.full_name.toLowerCase().includes(substr.toLowerCase())
+		|| image.targetdevice.title.toLowerCase().includes(substr.toLowerCase()) ? true : false
```
Comment (Gerasimov94) [inline src/common/selectors/entities.ts]: same mistake here

File: src/modules/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -52,33 +53,35 @@ export default class Dashboard extends PureComponent<IProps> {
...
 									activeImage={activeImage}
+									imagesByActiveStatus={imagesByActiveStatus}
+									textFilter={textFilter}
 								/>
-							</div>
-						)
-						: <ImagesPreloader text='loading images' />
-				}
+							) : (
+								<ImagesPreloader text='loading images' />
```
Comment (Gerasimov94) [inline src/modules/Dashboard/Dashboard.tsx]: `: <ImagesPreloader text='loading images' />`

File: src/modules/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -52,33 +53,35 @@ export default class Dashboard extends PureComponent<IProps> {
...
+							)
+						}
+					</div>
+					<CurrentImageTab
```
Comment (Gerasimov94) [inline src/modules/Dashboard/Dashboard.tsx]: in one row

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx [hunk 4]
```diff
@@ -48,8 +50,21 @@ export default class ImageCard extends Component<IProps, IState> {
 		image: {
 			thumb: '',
 		},
+		textFilter: '',
 	};
 
+	public findInTextWrapper = (text: string) => {
+		if (this.props.textFilter) {
```
Comment (Gerasimov94) [inline src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx:61]: multiple usage of props

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx [hunk 5]
```diff
@@ -154,7 +170,7 @@ export default class ImageCard extends Component<IProps, IState> {
 				</div>
 				<div className={styles.titles}>
 					<img className={styles.titles_img} src={ thumb } />
-					<div className={styles.titles_main}>{ distro }</div>
+					<div className={styles.titles_main}>{ findInTextWrapper(distro) }</div>
```
Comment (Gerasimov94) [inline src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx]: extra spaces in curly braces

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx (hunk 4 above)
Comment (Gerasimov94) [reply src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx:61]: i think you must use that inside selectors. here must be just an action creator, you type something, selector for items recalculates and return a new array of items.

Comment (Gerasimov94) [reply src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx:61]: i think it is critical. @CodeAnnihilator ?

**Review by Gerasimov94 — CHANGES_REQUESTED** [review]

**Comment by alena-andr** [comment]
change commit name

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx (hunk 4 above)
Comment (CodeAnnihilator) [reply src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx:61]: right

Comment (CodeAnnihilator) [reply src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx:61]: split this whole file on smaller logical components

**Review by CodeAnnihilator — CHANGES_REQUESTED** [review]
rename commit message according to our standards

**Review by CodeAnnihilator — CHANGES_REQUESTED** [review]
i don't see your comment with "done" on each required comment. Please fix that

File: src/common/selectors/entities.ts (hunk 1 above)
Comment (gusartus) [reply src/common/selectors/entities.ts]: done

File: src/modules/Dashboard/Dashboard.tsx (hunk 3 above)
Comment (gusartus) [reply src/modules/Dashboard/Dashboard.tsx]: done

File: src/modules/Dashboard/Dashboard.tsx (hunk 2 above)
Comment (gusartus) [reply src/modules/Dashboard/Dashboard.tsx]: done

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx (hunk 5 above)
Comment (gusartus) [reply src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx]: done

**Pull request closed without merging**